Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -846,9 +1008,13 @@ BTDiagnostics::Flush (int i_buffer)
             m_buffer_box[i_buffer].setSmall(m_moving_window_dir, (m_buffer_box[i_buffer].smallEnd(m_moving_window_dir) + 1) );
             m_buffer_box[i_buffer].setBig(m_moving_window_dir, (m_buffer_box[i_buffer].bigEnd(m_moving_window_dir) - 1) );
             m_particles_buffer[i_buffer][0]->SetParticleBoxArray(0,vba.back());
```

Comment:
In this function, `m_particles_buffer.at(i_buffer)` is accessed a bunch of times. It might be worth in a following clean-up to declare a reference to it somewhere further up.